src/main/java/io/cdap/plugin/google/common/GoogleAuthBaseConfig.java

```diff
@@ -238,4 +298,53 @@ public String getClientSecret() {
   public String getRefreshToken() {
     return refreshToken;
   }
+
+  @Nullable
+  public String getServiceAccountFilePath() {
+    if (containsMacro(ACCOUNT_FILE_PATH)  || accountFilePath == null ||
+      accountFilePath.isEmpty() || AUTO_DETECT_VALUE.equals(accountFilePath)) {
+      return null;
+    }
+    return accountFilePath;
+  }
+
+  @Nullable
+  public String getServiceAccountJson() {
+    if (containsMacro(NAME_SERVICE_ACCOUNT_JSON) || Strings.isNullOrEmpty(serviceAccountJson)) {
+      return null;
+    }
+    return serviceAccountJson;
+  }
+
+  /**
+   * @return Service Account Type, defaults to filePath.
+   */
+  @Nullable
+  public String getServiceAccountType() {
+    if (containsMacro(NAME_SERVICE_ACCOUNT_TYPE)) {
+      return null;
+    }
+    return Strings.isNullOrEmpty(serviceAccountType) ? SERVICE_ACCOUNT_FILE_PATH : serviceAccountType;
+  }
+
+  @Nullable
+  public Boolean isServiceAccountJson() {
+    String serviceAccountType = getServiceAccountType();
+    return Strings.isNullOrEmpty(serviceAccountType) ? null : serviceAccountType.equals(SERVICE_ACCOUNT_JSON);
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    String serviceAccountType = getServiceAccountType();
+    return Strings.isNullOrEmpty(serviceAccountType) ? null : serviceAccountType.equals(SERVICE_ACCOUNT_FILE_PATH);
+  }
+
+  @Nullable
+  public String getServiceAccount() {
```

Comment:
This is only used in GoogleBasicAuthConfig.java where you're checking if the returned string is null or empty. The string value is unused. 

It seems this is unnecessary. You just need something like this:
```
if (config.isServiceAccountFilePath()) {
...
} else if (config. isServiceAccountJson()) {
...
} else {
// use default credentials
}
```